src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -419,7 +434,7 @@ export class NodeClientRequest extends ClientRequest {
         const firstClone = cloneIncomingMessage(response)
         const secondClone = cloneIncomingMessage(response)
 
-        this.emit('response-internal', secondClone)
+        this.emit('response-internal', secondClone, firstClone)
```

Comment:
We need to expose the original `IncomingMessage` of the original response if we wish to delay its `end` event emission. This modifies the listener call signature of the `response-internal` event.